jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/FormFieldsTest.java

```diff
@@ -32,31 +34,44 @@
 import org.junit.jupiter.params.provider.MethodSource;
 
 import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.instanceOf;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
 public class FormFieldsTest
 {
-    public static Stream<Arguments> tests()
+    public static Stream<Arguments> validData()
     {
         return Stream.of(
+            Arguments.of(List.of(""), UTF_8, -1, -1, Map.of()),
+            Arguments.of(List.of("name"), UTF_8, -1, -1, Map.of("name", "")),
+            Arguments.of(List.of("name&"), UTF_8, -1, -1, Map.of("name", "")),
+            Arguments.of(List.of("name="), UTF_8, -1, -1, Map.of("name", "")),
+            Arguments.of(List.of("%A"), UTF_8, -1, -1, Map.of("%A", "")),
```

Comment:
single character pct-encoding?
This seems like a bug, and something we shouldn't support.